tests/unit/client-eviction.tcl

```diff
@@ -531,5 +531,87 @@ start_server {} {
     } {} {needs:debug}
 }
 
+start_server {} {
+    test "client total memory grows during client no-evict on" {
+        r client no-evict on
+
+        r setrange k [mb 1] v
+        set rr [redis_deferring_client]
+        set cname "test_client"
+        $rr client setname $cname
+        $rr client no-evict on
+        $rr flush
+
+        assert {[$rr read] == "OK"}
+
+        set maxmemory_clients 1
+        r config set maxmemory-clients $maxmemory_clients
+
+        # Fill output buffer in loop without reading it and make sure 
+        # the tot-mem of client has increased and eviction not occurring.
+        set old_client_mem [client_field test_client tot-mem]
+        set retry 10
+        while {$retry} {
+            $rr get k
+            incr retry -1
+        }
+        set mem [client_field test_client tot-mem]
+        assert {$mem > $old_client_mem}
+
+        # Trigger the client eviction, by flipping the no-evict flag to off
+        $rr client no-evict off
+        while true {
+            if { [catch {
+                $rr get k
+                $rr flush
+               } e]} {
+                assert {![client_exists test_client]}
+                break
```

Comment:
don't we expect it to get disconnected even if we don't add more GET commands?
i.e. maybe a simple `wait_for_condition` is enough here?